controllers/imageclusterinstall_controller.go

```diff
@@ -451,24 +452,13 @@ func (r *ImageClusterInstallReconciler) writeInputData(ctx context.Context, log
 			return err
 		}
 		clusterInfo := r.getClusterInfoFromFile(clusterInfoFilePath)
-		crypto := &lca_api.KubeConfigCryptoRetention{}
-		if clusterInfo == nil {
-			// handle first cert generation
-			crypto, err = r.generateClusterCrypto(ctx, cd)
-			if err != nil {
-				return err
-			}
-		} else {
-			crypto = &clusterInfo.KubeconfigCryptoRetention
-			if clusterInfo.ClusterName != cd.Spec.ClusterName || clusterInfo.BaseDomain != cd.Spec.BaseDomain {
-				// in case the cluster name or baseDomain changed we need a new kubeconfig, so we just regenerate all crypto
-				crypto, err = r.generateClusterCrypto(ctx, cd)
-				if err != nil {
-					return err
-				}
-			}
+		crypto, err := r.EnsureKubeconfigSecret(ctx, cd, clusterInfo)
+
+		kubeadminPasswordHash, err := r.Credentials.EnsureAdminPasswordSecret(ctx, cd)
```

Comment:
Can we be consistent and either use `r.Credentials` or omit the `Credentials`?

I'd prefer it not be embedded FWIW (i.e. have a distinct field called `Credentials` like you did for `CertManager`)